gr-uhd/apps/uhd_siggen_base.py

```diff
@@ -55,6 +57,16 @@ class USRPSiggen(gr.top_block, pubsub, UHDApp):
     def __init__(self, args):
         gr.top_block.__init__(self)
         pubsub.__init__(self)
+        if not 0.0 <= args.amplitude <= 1.0:
```

Comment:
technically: 0.0 < args.ampl ... not <= ... log(0) -> -inf ... which we want to avoid here ... this the "Must be" line would be "(0.0, 1.0]" to be math notationally correct ... yes?

Reply:
No, 0.0 is a valid amplitude for practical reasons (want to temporarily turn off the signal to see what spurs are coming from the LOs or other clocks, for example). [0.0, 1.0] is correct.
But you're right, when in power mode, we need a way of handling that. I think I'll follow your suggestions until someone has a better idea -- but only conditionally on power mode.

Reply:
OK implemented and tested that.